tests/pipeline/test_entity_processing_stage_composite.py

```diff
@@ -0,0 +1,335 @@
+assert events_storage is not None
+events_table_name = events_storage.get_table_writer().get_schema().get_table_name()
+
+events_table = Table(
+    events_table_name,
+    events_storage.get_schema().get_columns(),
+    storage_key=events_storage.get_storage_key(),
+    allocation_policies=events_storage.get_allocation_policies(),
+    final=False,
+    sampling_rate=None,
+    mandatory_conditions=events_storage.get_schema()
+    .get_data_source()
+    .get_mandatory_conditions(),
+)
+
+groups_ent = Entity(
+    EntityKey.GROUPEDMESSAGE, get_entity(EntityKey.GROUPEDMESSAGE).get_data_model()
+)
+groups_storage = get_storage(StorageKey.GROUPEDMESSAGES)
+groups_schema = groups_storage.get_schema()
+assert isinstance(groups_schema, TableSchema)
+
+groups_table = Table(
+    groups_schema.get_table_name(),
+    groups_schema.get_columns(),
+    storage_key=groups_storage.get_storage_key(),
+    allocation_policies=groups_storage.get_allocation_policies(),
+    final=False,
+    sampling_rate=None,
+    mandatory_conditions=groups_schema.get_data_source().get_mandatory_conditions(),
+)
+
+TEST_CASES = [
+    pytest.param(
+        CompositeQuery(
+            from_clause=LogicalQuery(
+                from_clause=events_ent,
+                selected_columns=[
+                    SelectedExpression("project_id", Column(None, None, "project_id")),
+                    SelectedExpression(
+                        "count_environment",
+                        FunctionCall(
+                            "count_environment",
+                            "uniq",
+                            (
+                                SubscriptableReference(
+                                    None,
+                                    Column(None, None, "tags"),
+                                    Literal(None, "environment"),
+                                ),
+                            ),
+                        ),
+                    ),
+                ],
+                groupby=[Column(None, None, "project_id")],
+                condition=binary_condition(
+                    BooleanFunctions.AND,
+                    binary_condition(
+                        ConditionFunctions.EQ,
+                        Column(None, None, "project_id"),
+                        Literal(None, 1),
+                    ),
+                    binary_condition(
+                        ConditionFunctions.GTE,
+                        Column(None, None, "timestamp"),
+                        Literal(None, datetime(2020, 1, 1, 12, 0)),
+                    ),
+                ),
+            ),
+            selected_columns=[
+                SelectedExpression(
+                    "average",
+                    FunctionCall(
+                        "average", "avg", (Column(None, None, "count_environment"),)
+                    ),
+                ),
+            ],
+        ),
+        CompositeQuery(
```

Comment:
unrelated to this PR but i think its confusing that physical composite and logical composite are both instances of the same class.